File: malevich-web/src/app/_transfer/exchangeOrderDto.ts
import {PaymentMethodDto} from "./paymentMethodDto";

export class ExchangeOrderDto {
  id: number;

  paymentMethod: PaymentMethodDto;

  exchangeName: string;

  internalStatus: string;

  type: string;

  originalAmount: number;

  currencyPair: string;

  orderId: string;

  timestamp: Date;

  status: string;

  cumulativeAmount: number;

  averagePrice: number;

  fee: number;

  leverage: string;
}

File: malevich-server/src/main/java/io/malevich/server/transfer/ExchangeOrderDto.java
package io.malevich.server.transfer;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
public class ExchangeOrderDto {

    private long id;

    private PaymentMethodSuperDto paymentMethod;

    private String exchangeName;

    private String internalStatus;

    private String type;

    private double originalAmount;

    private String currencyPair;

    private String orderId;

    @JsonFormat(pattern = "dd/MM/yyyy")
    private java.sql.Timestamp timestamp;

    private String status;

    private double cumulativeAmount;

    private double averagePrice;

    private double fee;

    private String leverage;

}
